### PR TITLE
MTDSA-8541 Added NRS calls to Submit UK Prop Bsas

### PR DESCRIPTION
**Pull request by HarryBaxter**

### CODE REVIEW

File: app/v1/controllers/SubmitUkPropertyBsasController.scala [hunk 1]
```diff
@@ -67,10 +68,14 @@ class SubmitUkPropertyBsasController @Inject()(val authService: EnrolmentsAuthSe
       val featureSwitch = FeatureSwitch(appConfig.featureSwitch)
 
       val rawData = SubmitUkPropertyBsasRawData(nino, bsasId, AnyContentAsJson(request.body))
+
       val result =
         for {
           parsedRequest <- EitherT.fromEither[Future](requestParser.parseRequest(rawData))
           response <- {
+            //Submit asynchronously to NRS
+            nrsService.submit(nino, request.body.as[SubmitUKPropertyBsasRequestBody])
```
Comment (james-work-account) [inline app/v1/controllers/SubmitUkPropertyBsasController.scala:77]: could these be taken from the parsedRequest instead? like `nrsService.submit(nino, parsedRequest.body)` or whatever it's called

File: app/v2/controllers/SubmitUkPropertyBsasController.scala [hunk 2]
```diff
@@ -63,10 +64,16 @@ class SubmitUkPropertyBsasController @Inject()(val authService: EnrolmentsAuthSe
...
 
       val rawData = SubmitUkPropertyBsasRawData(nino, bsasId, AnyContentAsJson(request.body))
+
       val result =
         for {
           parsedRequest <- EitherT.fromEither[Future](requestParser.parseRequest(rawData))
-          response      <- EitherT(service.submitPropertyBsas(parsedRequest))
+          response      <- {
+            //Submit asynchronously to NRS
+            nrsService.submit(nino, request.body.as[SubmitUKPropertyBsasRequestBody])
```
Comment (james-work-account) [inline app/v2/controllers/SubmitUkPropertyBsasController.scala:73]: as above

File: app/v1/controllers/SubmitUkPropertyBsasController.scala (hunk 1 above)
Comment (HarryBaxter) [reply app/v1/controllers/SubmitUkPropertyBsasController.scala:77]: I'll look into it

Comment (HarryBaxter) [reply app/v1/controllers/SubmitUkPropertyBsasController.scala:77]: That seems to be fine,I'll push up the changes

**Review by Mbrookesy — APPROVED** [review]

File: app/v1/controllers/SubmitUkPropertyBsasController.scala [hunk 3]
```diff
@@ -30,16 +30,17 @@ import v1.controllers.requestParsers.SubmitUkPropertyBsasDataParser
 import v1.hateoas.HateoasFactory
 import v1.models.audit.{AuditEvent, AuditResponse, GenericAuditDetail}
 import v1.models.errors.{FormatAdjustmentValueError, RuleAdjustmentRangeInvalid, _}
-import v1.models.request.submitBsas.SubmitUkPropertyBsasRawData
+import v1.models.request.submitBsas.{SubmitUkPropertyBsasRawData, SubmitUKPropertyBsasRequestBody}
```
Comment (Mbrookesy) [inline app/v1/controllers/SubmitUkPropertyBsasController.scala:33]: unused import

File: app/v2/controllers/SubmitUkPropertyBsasController.scala [hunk 4]
```diff
@@ -29,15 +29,16 @@ import v2.controllers.requestParsers.SubmitUkPropertyBsasDataParser
 import v2.hateoas.HateoasFactory
 import v2.models.audit.{AuditEvent, AuditResponse, GenericAuditDetail}
 import v2.models.errors.{FormatAdjustmentValueError, RuleAdjustmentRangeInvalid, _}
-import v2.models.request.submitBsas.ukProperty.SubmitUkPropertyBsasRawData
+import v2.models.request.submitBsas.ukProperty.{SubmitUkPropertyBsasRawData, SubmitUKPropertyBsasRequestBody}
```
Comment (Mbrookesy) [inline app/v2/controllers/SubmitUkPropertyBsasController.scala:32]: unused import

**Review by Mbrookesy — CHANGES_REQUESTED** [review]

**Review by Mbrookesy — APPROVED** [review]

**Pull request merged**